p2p/module.go

```diff
@@ -0,0 +1,193 @@
+package p2p
+
+import (
+	"errors"
+	"log"
+	"net"
+	"pocket/p2p/types"
+	"pocket/shared/config"
+	"pocket/shared/modules"
+	"pocket/shared/types"
+	"sync"
+
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	MaxInbound           uint = 128
+	MaxOutbound          uint = 128
+	WireByteHeaderLength int  = 9
+	ReadBufferSize       int  = (1024 * 4)
+	WriteBufferSize      int  = (1024 * 4)
+	ReadDeadlineMs       int  = 400
+)
+
+var (
+	Protocol = "tcp"
+	Address  = "localhost:30303"
+)
+
+type P2PModule struct {
+	modules.NetworkModule
+
+	bus    modules.Bus
+	config *config.P2PConfig
+	c      *dcodec
+
+	id           uint64
+	protocol     string
+	address      string
+	externaladdr string
+
+	inbound  pipemap
+	outbound pipemap
+
+	peerlist *types.Peerlist
+
+	listener struct {
+		sync.Mutex
+		*net.TCPListener
+	}
+	listening atomic.Bool
+
+	done    chan uint
```

Comment:
1. Consider create a type for `chan uint`.
2. It's unclear to me why all of these are unit channels. Consider adding a comment for what they're used for.

Reply:
They are for signaling primarily for execution and what not. When we'll start using `context.Context`, this could be replaced with a cancellable context.